Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp

```diff
@@ -374,7 +374,7 @@ void RemoteRenderingBackend::releaseRemoteResourceWithQualifiedIdentifier(Qualif
             remoteDisplayList->clearImageBufferReference();
     }
     auto success = m_remoteResourceCache.releaseRemoteResource(renderingResourceIdentifier);
-    MESSAGE_CHECK(success, "Resource is being released before being cached.");
+    MESSAGE_CHECK(success || !m_streamConnection->connection().isValid(), "Resource is being released before being cached.");
```

Comment:
This is not correct.
The message can be processed at arbitrary time.
The connection may be invalidated at arbitrary time.

It is valid to dispatch the "release remote resource" message and get the connection invalidated during the dispatch.

Reply:
I changed m_remoteResourceCache to be a unique_ptr. I set it to null in the clearing block and I check it before using it. So adding to it or removing from it will not happen once it is cleared.

Clearing, checking and accessing the m_remoteResourceCache happen only on the WorkQueue. So there should not be a problem due to ordering or synchronization here.